vars/mavenExecute.groovy

```diff
@@ -25,3 +29,39 @@ def call(Map parameters = [:]) {
     }
     return output
 }
+
+private void validateParameter(parameter, name) {
+    if (!parameter) {
+        return
+    }
+
+    String errorMessage = "Expected parameter ${name} with value ${parameter} to be of type List, but it is ${parameter.class}. "
+
+    // Specifically check for string-like types as the old step (v1.23.0 and before) allowed that as input
+    if (parameter in CharSequence) {
+        errorMessage += "This is a breaking change for mavenExecute in library version v1.24.0 which allowed strings as input for defines, flags and goals before. " +
+            "To fix that, please update the interface to pass in lists, or use v1.23.0 which is the last version with the old interface. "
+
+        if (parameter.contains(BashUtils.ESCAPED_SINGLE_QUOTE)) {
+            errorMessage += "It looks like your input contains shell escaped quotes. "
+        }
+
+        error errorMessage + "Note that *no* shell escaping is allowed."
+    }
+
+    if (!parameter in List) {
+        error errorMessage + "Note that *no* shell escaping is allowed."
+    }
+
+    for (int i = 0; i < parameter.size(); i++) {
+        String element = parameter[i]
+
+        if (element =~ /-D.*='.*'/) {
+            echo "[$STEP_NAME WARNING] It looks like you passed a define in the form -Dmy.key='this is my value' in $element. Please note that the quotes might cause issues. Correct form: -Dmy.key=this is my value"
+        }
+
+        if (element.startsWith("'") && element.endsWith("'") && element.contains(' ')) {
+            echo "[$STEP_NAME WARNING] It looks like $element is quoted but it should not be quoted."
+        }
```

Comment:
Do you think there is a situation where a define in the form of `-Dmy.key='this is my value'` or even just `-Dmy.key='value'` (without spaces, but quotes regardless) is actually valid? Same for any other quoted entry. Especially the second check would not catch cases where quotes were added, but were not strictly necessary (due to third condition).